compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2117,7 +2117,13 @@ impl<'tcx> TyCtxt<'tcx> {
                     _ => bug!("unexpected parent item of associated item: {parent_def_id:?}"),
                 }
             }
-            DefKind::Closure | DefKind::OpaqueTy => {
+            DefKind::OpaqueTy => match self.opaque_ty_origin(def_id) {
+                hir::OpaqueTyOrigin::FnReturn { parent, .. } => self.is_conditionally_const(parent),
```

Comment:
Am I understanding this correctly that this would treat the `impl Trait` in

```rust
const fn a() -> impl Trait {}
```

as conditionally const? If so, then this is wrong, and the `~const`ness of an opaque type should be recorded in some way.

Reply:
No, you're misreading. It's conditionally const, but it just has no const item bounds.

Reply:
Got it, thanks